drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1331,7 +1331,9 @@ static void eth_iface_init(struct net_if *iface)
 		k_thread_create(&dev_data->rx_thread, dev_data->rx_thread_stack,
 				K_KERNEL_STACK_SIZEOF(dev_data->rx_thread_stack),
 				rx_thread, (void *) dev, NULL, NULL,
-				K_PRIO_COOP(CONFIG_ETH_STM32_HAL_RX_THREAD_PRIO),
+				IS_ENABLED(CONFIG_NET_TC_THREAD_PREEMPTIVE)
```

Comment:
you should probably make a separate driver kconfig for this. If you think the two configs will most likely have the same value then you could  `default y if NET_TC_THREAD_PREEMPTIVE`

Reply:
> you should probably make a separate driver kconfig for this. If you think the two configs will most likely have the same value then you could `default y if NET_TC_THREAD_PREEMPTIVE`

There is already a KConfig for hal_stm32_ethernet: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/ethernet/Kconfig.stm32_hal Thus it is not necessary I think. Proper description of the behavior in help section of KConfig would be fine.

Reply:
That's not what I'm talking about, I mean a Kconfig to say if the driver rx thread priority is preemptive, instead of coupling it to the net tc thread priority being preemptive

Reply:
Oh, I see. It makes sense. Thanks.

Reply:
> you should probably make a separate driver kconfig for this. 

I'm not sure why this would be required. `ETH_STM32_HAL_RX_THREAD_PRIO` is neutral.
And a STM32 specific symbol to reflect value of subsystem's one doesn't add much aside a possibility of misconfiguration.

Reply:
I think it is more about NET_TC_THREAD_PREEMPTIVE

Reply:
Does it mean adding ETH_STM32_HAL_RX_THREAD_PREEMPTIVE to drivers/ethernet/Kconfig.stm32_hal ?

```shell
config ETH_STM32_HAL_RX_THREAD_PREEMPTIVE
    bool "STM32 Ethernet RX Thread Preemptive"
    default y if NET_TC_THREAD_PREEMPTIVE
    help
      Enable preemptive scheduling for the RX thread of STM32 Ethernet.
``` 
...and changing the thread creation?

```suggestion
				IS_ENABLED(CONFIG_ETH_STM32_HAL_RX_THREAD_PREEMPTIVE)
				
```

Reply:
> I think it is more about NET_TC_THREAD_PREEMPTIVE

Yes, my response was toward this proposal indeed.

Reply:
> Does it mean adding ETH_STM32_HAL_RX_THREAD_PREEMPTIVE to drivers/ethernet/Kconfig.stm32_hal ?
> 
> ```shell
> config ETH_STM32_HAL_RX_THREAD_PREEMPTIVE
>     bool "STM32 Ethernet RX Thread Preemptive"
>     default y if NET_TC_THREAD_PREEMPTIVE
>     help
>       Enable preemptive scheduling for the RX thread of STM32 Ethernet.
> ``` 
> ...and changing the thread creation?
> 
> ```suggestion
> 				IS_ENABLED(CONFIG_ETH_STM32_HAL_RX_THREAD_PREEMPTIVE)
> 				
> ```

@decsny I think, it could be better, if you clarify it.